### DISCUSSION TITLE
Logging All request headers from incoming requests in Keycloak 17.x

### DISCUSSION THREAD

**Discussion by AndreiMihaiRad** [keycloak/keycloak#11221] [Archive]
What is the recommended way to intercept and log request headers from incoming traffic.
I need this for debugging purposes because I run Keycloak behind an edge proxy and I want to see if headers are correct passed between client and server. 

PS: In past version we implemented HttpHandler interface, but that was an io.undertow:undertow-core dependency.

**Comment by datacoderid** [comment]
Hi, I face same issue and found the solution for this

1. Create quarkus.properties in /conf
2. Add followings records:
quarkus.http.access-log.enabled=true
quarkus.http.access-log.pattern=%r\n%{ALL_REQUEST_HEADERS}
3. Rebuild kc and start

**Reply by kkcmadhu** [reply]
thats true. it works. but i found there is better way.
Simply  pass these as  env variable to your  docker with -e, this will work in both docker and k8s.
you will have better control and need not build a new image every time you want to change quarkus.properties.

not sure why keycloak folks havent documented this clearly..

**Reply by kkcmadhu** [reply]
> 2\. quarkus.http.access-log.enabled=true

well.. the defaults can be in quarkus.properties and env variables can alway override them.. by personal preference is env variable,as you can change things at run time, without getting your image built and deployed again.

**Reply by bilak** [reply]
few years later ... I also needed this. In docker (compose) it's just enough to set env variables:
```
QUARKUS_HTTP_ACCESS_LOG_ENABLED: true
QUARKUS_HTTP_ACCESS_LOG_LOG_DIRECTORY: /tmp
QUARKUS_HTTP_ACCESS_LOG_LOG_TO_FILE: true
QUARKUS_HTTP_ACCESS_LOG_PATTERN: "%r\n%{ALL_REQUEST_HEADERS}"
```
maybe it will help someone in the future

**Comment by mohamedbayoudh** [comment]
Hi guys, maybe you can help me: 

I am running keycloak behind traefik in a docker container. I am trying to inspect the incoming header requests to my keycloak docker container by enabling quarkus access log. I tried out the approach of @datacoderid but did not help. (Version 22.0.4) 
Is there another log file for quarkus ?
